internal/provider/resource_gitlab_group_share_group_test.go

```diff
@@ -125,46 +104,50 @@ func testAccCheckGitlabGroupIsNotShared(groupName string) resource.TestCheckFunc
 	}
 }
 
-func testAccGitlabGroupShareGroupConfig(
-	randName string,
-	shareGroupSettings string,
-) string {
-	return fmt.Sprintf(
-		`
-		resource "gitlab_group" "test_main" {
-		  name = "%[1]s_main"
-		  path = "%[1]s_main"
+func testAccCheckGitlabShareGroupDestroy(s *terraform.State) error {
+	var groupId string
+	var sharedGroupId int
+	var err error
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type == "gitlab_group_share_group" {
+			groupId, sharedGroupId, err = groupIdsFromId(rs.Primary.ID)
+			if err != nil {
+				return fmt.Errorf("[ERROR] cannot get Group ID and ShareGroupId from input: %v", rs.Primary.ID)
+			}
 		}
+	}
 
-		resource "gitlab_group" "test_share" {
-		  name = "%[1]s_share"
-		  path = "%[1]s_share"
-		}
+	// Get Main Group
+	group, _, err := testGitlabClient.Groups.GetGroup(groupId, nil)
```

Comment:
This will fail if the `for` doesn't find the resource ...

Reply:
Done